CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+        {
+            configWindow.Dispatcher.Invoke(() => { BindingOperations.DisableCollectionSynchronization(criminalrecord); });
+        }
+
+        public void HandleProfile(JObject profile)
+        {
+        }
+
+        public void PostHandle(Event @event)
+        {
+        }
+
+        public void PreHandle(Event @event)
+        {
+            Logging.Debug("Received event " + JsonConvert.SerializeObject(@event));
+
+            // Handle the events that we care about
+            if (@event is BondAwardedEvent)
+            {
+                handleBondAwardedEvent((BondAwardedEvent)@event);
+            }
+            else if (@event is BondRedeemedEvent)
+            {
+                handleBondRedeemedEvent((BondRedeemedEvent)@event);
+            }
+            else if (@event is BountyAwardedEvent)
+            {
+                handleBountyAwardedEvent((BountyAwardedEvent)@event);
+            }
+            else if (@event is BountyIncurredEvent)
+            {
+                handleBountyIncurredEvent((BountyIncurredEvent)@event);
+            }
+            else if (@event is BountyPaidEvent)
+            {
+                handleBountyPaidEvent((BountyPaidEvent)@event);
+            }
+            else if (@event is BountyRedeemedEvent)
+            {
+                handleBountyRedeemedEvent((BountyRedeemedEvent)@event);
+            }
+            else if (@event is FineIncurredEvent)
+            {
+                handleFineIncurredEvent((FineIncurredEvent)@event);
+            }
+            else if (@event is FinePaidEvent)
+            {
+                handleFinePaidEvent((FinePaidEvent)@event);
+            }
+            else if (@event is DiedEvent)
+            {
+                handleDiedEvent((DiedEvent)@event);
+            }
+
+        }
+
+        private void handleBondAwardedEvent(BondAwardedEvent @event)
+        {
+            if (@event.timestamp > updateDat)
+            {
+                updateDat = @event.timestamp;
+                _handleBondAwardedEvent(@event);
+                writeRecord();
+            }
+        }
+
+        private void _handleBondAwardedEvent(BondAwardedEvent @event)
+        {
+            int shipId = EDDI.Instance?.CurrentShip?.LocalId ?? 0;
+            string currentSystem = EDDI.Instance?.CurrentStarSystem?.name;
+            FactionReport report = new FactionReport(@event.timestamp, false, shipId, Crime.None, currentSystem, @event.reward)
+            {
+                station = EDDI.Instance?.CurrentStation?.name,
+                body = EDDI.Instance?.CurrentStellarBody?.name,
+                victim = @event.victimfaction
+            };
+
+            FactionRecord record = GetRecordWithFaction(@event.awardingfaction);
+            if (record == null)
+            {
+                record = AddRecord(@event.awardingfaction);
+            }
+            record.factionReports.Add(report);
+            record.claims += @event.reward;
```

Comment:
Code similar to 
```cs
	            FactionReport report = new FactionReport(@event.timestamp, false, shipId, Crime.None, currentSystem, @event.reward)
            {
                station = EDDI.Instance?.CurrentStation?.name,
                body = EDDI.Instance?.CurrentStellarBody?.name,
                victim = @event.victimfaction
            };

             FactionRecord record = GetRecordWithFaction(@event.awardingfaction);
            if (record == null)
            {
                record = AddRecord(@event.awardingfaction);
            }
            record.factionReports.Add(report);
            record.claims += @event.reward;
```
comes up a number of times in this file. Consider refactoring to minimize redundancy?

Reply:
I tried. It was clunky and didn't like it.